WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1317,10 +1335,7 @@ extension EditableOrderViewModel {
                     self.analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: self.flow.analyticsFlow,
                                                                                     source: .orderCreation,
                                                                                     addedVia: .scanning))
-                    // The scanned product or variation was added locally when it was found
-                    // Let's refresh our references so we can retrieve it
-                    self.updateLocalItemsReferences()
```

Comment:
We don't need to refresh the storage when a new product is found after scanning. We will add it on `updateOrderWithProduct`. Furthermore, we don't want to sync products from the database, as it would remove added products/variations that are not in the DB